monitoring/uss_qualifier/scenarios/astm/utm/dss/oir_explicit_sub_handling.py

```diff
@@ -0,0 +1,521 @@
+                    )
+        self.end_test_step()
+
+    def _step_update_oir_with_sufficient_explicit_sub(self, is_replacement: bool):
+        step_name = (
+            "Replace the OIR's explicit subscription"
+            if is_replacement
+            else "Attach explicit subscription to OIR"
+        )
+        self.begin_test_step(step_name)
+        self._current_oir, _, _ = oir_fragments.update_oir_query(
+            scenario=self,
+            dss=self._dss,
+            oir_id=self._oir_id,
+            ovn=self._current_oir.ovn,
+            oir_params=self._planning_area.get_new_operational_intent_ref_params(
+                key=[],
+                state=OperationalIntentState.Accepted,
+                uss_base_url=self._planning_area.get_base_url(),
+                time_start=self._current_extra_sub.time_start.value.datetime,
+                time_end=self._current_extra_sub.time_end.value.datetime,
+                subscription_id=self._extra_sub_id,
+            ),
+        )
+        self.end_test_step()
+
+    def _step_oir_has_correct_subscription(
+        self, expected_sub_id: Optional[SubscriptionID]
+    ):
+        step_check_name = (
+            "OIR is not attached to any subscription"
+            if expected_sub_id is None
+            else "OIR is attached to expected subscription"
+        )
+        self.begin_test_step(step_check_name)
+        with self.check("Get operational intent reference by ID", self._pid) as check:
+            try:
+                oir, q = self._dss.get_op_intent_reference(self._oir_id)
+                self.record_query(q)
+            except QueryError as qe:
+                self.record_queries(qe.queries)
+                check.record_failed(
+                    summary="Could not get OIR",
+                    details=f"Failed to get OIR with error code {qe.cause_status_code}: {qe.msg}",
+                    query_timestamps=qe.query_timestamps,
+                )
+
+        sub_is_as_expected = False
+        referenced_sub_was_found_when_non_expected = False
+        if expected_sub_id is None:
+            # The official DSS implementation will set the subscription ID to 00000000-0000-4000-8000-000000000000 when the OIR is not attached to any subscription.
+            # Other implementations may use a different value, as the OpenAPI spec does not allow the value to be empty
+            # We may at some point decide to tolerate accepting empty returned values here,
+            # but in the meantime we simply attempt to obtain the subscription and check that it does not exist
+            if oir.subscription_id == NULL_SUBSCRIPTION_ID:
+                # Sub ID explicitly set to the value representing the null subscription: all good
+                sub_is_as_expected = True
+            elif oir.subscription_id is None:
+                # Sub ID not set at all: not strictly compliant with the spec, but acceptable in this context
```

Comment:
Why is this acceptable in this context?  Maybe:

```suggestion
                # Sub ID not set at all: not compliant with the spec, but not wrong with regard to which subscription should be attached to the OIR
```